src/sentry_core.c

```diff
@@ -785,4 +796,23 @@ sentry_transaction_finish(sentry_value_t tx)
     // This takes ownership of the transaction, generates an event ID, merges
     // scope
     return sentry__capture_event(tx);
+
+fail:
+    sentry_value_decref(tx);
+    return sentry_uuid_nil();
+}
+
+sentry_value_t
+sentry_set_span(sentry_value_t transaction)
+{
+    sentry_uuid_t nil_uuid = sentry_uuid_nil();
+    sentry_value_t sentinel = sentry__value_new_internal_uuid(&nil_uuid);
+    SENTRY_WITH_SCOPE_MUT (scope) {
+        sentry_value_decref(sentinel);
+        sentinel = sentry__value_clone(
```

Comment:
instead of cloning, you can just get another reference to it.
Also above, instead of creating a sentinel just to throw it away afterwards is a bit weird. Might as well just return a `null` value in that case.

Reply:
`sentry__value_clone` uses `sentry_value_incref` if the value is a string. wouldn't cloning here be equivalent to just getting another reference?